src/pkg/build/build.go

```diff
@@ -50,32 +58,46 @@ type Build struct {
 }
 
 // NewBuild creates a new Build struct from a spec (URI, definition file, etc...)
-func NewBuild(spec, dest, format string) (*Build, error) {
+func NewBuild(spec, dest, format string, force, update bool, sections []string, noTest bool) (*Build, error) {
 	def, err := makeDef(spec)
 	if err != nil {
 		return nil, fmt.Errorf("unable to parse spec %v: %v", spec, err)
 	}
 
-	return newBuild(def, dest, format)
+	return newBuild(def, dest, format, force, update, sections, noTest)
 }
 
 // NewBuildJSON creates a new build struct from a JSON byte slice
-func NewBuildJSON(r io.Reader, dest, format string) (*Build, error) {
+func NewBuildJSON(r io.Reader, dest, format string, force, update bool, sections []string, noTest bool) (*Build, error) {
 	def, err := types.NewDefinitionFromJSON(r)
 	if err != nil {
 		return nil, fmt.Errorf("unable to parse JSON: %v", err)
 	}
 
-	return newBuild(def, dest, format)
+	return newBuild(def, dest, format, force, update, sections, noTest)
 }
 
-func newBuild(d types.Definition, dest, format string) (*Build, error) {
+func newBuild(d types.Definition, dest, format string, force, update bool, sections []string, noTest bool) (*Build, error) {
+	var err error
```

Comment:
In here should do:

```
if update {
    format = "sandbox"
}
```

so users don't have to run `singularity build --update --sandbox ...`. It's already checked for in `cli/build.go` that the destination is a directory, so it's already implied.